wpilibNewCommands/src/main/native/include/frc2/command/ProxyCommand.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <memory>
+#include <span>
+
+#include <wpi/FunctionExtras.h>
+
+#include "frc2/command/CommandBase.h"
+#include "frc2/command/CommandHelper.h"
+#include "frc2/command/SetUtilities.h"
+
+namespace frc2 {
+/**
+ * Schedules the given command when this command is initialized, and ends when
+ * it ends. Useful for forking off from CommandGroups. If this command is
+ * interrupted, it will cancel the command.
+ *
+ * <p>This class is provided by the NewCommands VendorDep
+ */
+class ProxyCommand : public CommandHelper<CommandBase, ProxyCommand> {
+ public:
+  /**
+   * Creates a new ProxyCommand that schedules the supplied command when
+   * initialized, and ends when it is no longer scheduled. Useful for lazily
+   * creating commands at runtime.
+   *
+   * @param supplier the command supplier
+   */
+  explicit ProxyCommand(wpi::unique_function<Command*()> supplier);
+
+  /**
+   * Creates a new ProxyCommand that schedules the given command when
+   * initialized, and ends when it is no longer scheduled.
+   *
+   * @param command the command to run by proxy
+   */
+  explicit ProxyCommand(Command* command);
+
+  /**
+   * Creates a new ProxyCommand that schedules the given command when
+   * initialized, and ends when it is no longer scheduled.
+   *
+   * <p>Note that this constructor passes ownership of the given command to the
+   * returned ProxyCommand.
+   *
+   * @param command the command to schedule
+   */
+  explicit ProxyCommand(std::unique_ptr<Command>&& command);
```

Comment:
Typically in C++, you should take unique_ptr by value instead of by rvalue reference.  Taking by value implies you are transferring ownership, which is the case here.

Reply:
Probably.  But let's make that a separate PR.